apps/hash-frontend/src/pages/shared/org-form.tsx

```diff
@@ -0,0 +1,235 @@
+
+  const innerSubmit = handleSubmit(async (data) => {
+    try {
+      setLoading(true);
+      await onSubmit(data);
+    } catch (err) {
+      setSubmissionError(
+        typeof err === "string" ? err : (err as Error).message,
+      );
+    } finally {
+      setLoading(false);
+    }
+  });
+
+  return (
+    <Box component="form" onSubmit={innerSubmit} sx={{ background: "white" }}>
+      <InputGroup>
+        <Label
+          label="Display name"
+          hint="The primary name of your org, as it should appear on HASH"
+          htmlFor="name"
+          required
+        />
+        <TextField
+          autoFocus
+          error={!!nameError}
+          id="name"
+          helperText={nameError}
+          placeholder="e.g. Acme Corp"
+          {...register("name", { required: true })}
+          sx={{ width: 300 }}
+        />
+      </InputGroup>
+      <InputGroup>
+        <Label
+          label="Username"
+          hint="Once set, an org's username cannot be changed"
+          htmlFor="shortname"
+          required
+        />
+        <Box style={{ position: "relative" }}>
+          <Controller
+            control={control}
+            name="shortname"
+            rules={{
+              validate: validateShortname,
+            }}
+            render={({ field }) => (
+              <TextField
+                autoComplete="off"
+                disabled={!!initialOrg}
+                error={!!shortnameError}
+                helperText={shortnameError}
+                id="shortname"
+                inputProps={{
+                  sx: {
+                    borderColor: shortnameError ? "#FCA5A5" : "initial",
+                    "&:focus": {
+                      borderColor: shortnameError ? "#EF4444" : "initial",
+                    },
+                    paddingLeft: "2.25rem",
+                  },
+                }}
+                onBlur={field.onBlur}
+                onChange={(evt) => {
+                  const newEvt = { ...evt };
+                  newEvt.target.value = parseShortnameInput(
+                    newEvt.target.value,
+                  );
+                  field.onChange(newEvt);
+                }}
+                placeholder="acme"
+                sx={{ width: 300 }}
+              />
+            )}
+          />
+          <span
+            style={{
+              position: "absolute",
+              left: "1rem",
+              top: "1.5rem",
+              transform: "translateY(-50%)",
+              color: "#9CA3AF",
+            }}
+          >
+            @
+          </span>
+        </Box>
+      </InputGroup>
+      <InputGroup>
+        <Label
+          label="Website"
+          hint="Provide a link to help others identify your org"
+          htmlFor="website"
+        />
+        <TextField
+          id="name"
+          placeholder="https://acme.com"
+          sx={{ width: 400 }}
+          {...register("website", { required: false })}
+        />
```

Comment:
Maybe adding some basic validation such as `pattern="https?://.+"` could help ensure this URL to be in the expected format, not sure this is a big deal though.

Reply:
We should do this – I'll do it in a follow up when I add the company website to their profile page. Tracking in H-182